paddle/py_paddle/dataprovider_converter.py

```diff
@@ -109,7 +109,10 @@ def pre_scan(self, dat):
             if len(self.__shape__) > 3:
                 raise ValueError(
                     "The dimension of input cannot be greater than 3.")
-            self.__dim__ = reduce(lambda x, y: x * y, self.__shape__)
+            if len(self.__shape__) == 0:
+                self.__dim__ = 1
```

Comment:
The `DenseScanner` must accept the data type `dense_vector`. So the `dat` in line [108](https://github.com/PaddlePaddle/Paddle/pull/2636/files#diff-9f072c3613e2be3686afec9373b90059R108) must be a list (`[]`) or `numpy.array`. See the [input type doc](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/api/v1/data_provider/pydataprovider2_en.rst#input_types).

The format of input data in [issue](https://github.com/PaddlePaddle/Paddle/issues/2564) is not correct.